File: App/Questionnaire/src/main/java/com/example/mentalhealth/aiService/DTO/RequestMetadataDTO.java
package com.example.mentalhealth.aiService.DTO;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.Instant;
@Data
@AllArgsConstructor
@NoArgsConstructor
public class RequestMetadataDTO {

    private String requestId;              // traceability
    private String questionnaireId;         // psqi, phq9, gad7, etc.
    private String questionnaireName;       // optional, human readable
    private String language;                // "en", future i18n
    private Instant createdAt;
}

File: App/AI-service/models/RequestMetadataDTO.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime


class RequestMetadataDTO(BaseModel):
    requestId: str = Field(..., description="Traceability ID")
    questionnaireId: str = Field(..., description="gad7, phq9, psqi etc")
    questionnaireName: Optional[str] = Field(
        None, description="Human readable questionnaire name"
    )
    language: str = Field(..., description="Language code e.g. en")
    createdAt: datetime
    
    class Config:
        extra = "forbid"
